### PR TITLE
put-setter-values function

### PR DESCRIPTION
**Pull request by phanimarupaka**
Introduce a new function put-setter-values to the catalog.

### CODE REVIEW

File: examples/put-setter-values-simple/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+In this example, we will see how to put desired setter values in the `setters.yaml`
+file referenced from `Kptfile` via `configPath` option.
+
+### Fetch the example package
+
+Get the example package by running the following commands:
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-catalog.git/examples/put-setter-values-simple
```
Comment (frankfarzan) [inline examples/put-setter-values-simple/README.md:13]: are the packages versioned correctly post-merge?

File: functions/go/put-setter-values/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# put-setter-values
+
+### Overview
+
+<!--mdtogo:Short-->
+
+Put setter values in Kptfile or functionConfig for setters.
```
Comment (frankfarzan) [inline functions/go/put-setter-values/README.md:7]: I would rephrase this for more clarity. Also the fact the the functionConfig is inlined in Kptfile or not shouldn't be the important thing, so may want to remove that.

File: functions/go/put-setter-values/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+<!--mdtogo-->
+
+### FunctionConfig
+
+<!--mdtogo:Long-->
+
+This function is a porcelain to declare the setter values using imperative CLI command.
+This function is meant to be used as an alternative to editing file directly to declare
```
Comment (frankfarzan) [inline functions/go/put-setter-values/README.md:16]: "edit the file" Be more precise, what file are you talking about?

File: functions/go/put-setter-values/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    - stage
+```
+
+Invoke the function:
+
+```shell
+$ kpt fn eval -i gcr.io/kpt-fn/apply-setters:unstable -- image=ubuntu namespace=my-space env='[dev, prod]'
+```
+
+Modified setters.yaml looks like the following:
```
Comment (frankfarzan) [inline functions/go/put-setter-values/README.md:71]: setters.yaml -> `setters.yaml`

File: functions/go/put-setter-values/putsettervalues/putsettervalues.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+	FilePath string
+
+	// Name is name of the setter
+	Name string
+
+	// Value of the matching field
+	Value string
+}
+
+// Filter implements Set as a yaml.Filter
```
Comment (frankfarzan) [inline functions/go/put-setter-values/putsettervalues/putsettervalues.go:44]: comment on what this function does

File: functions/go/put-setter-values/putsettervalues/putsettervalues_test.go [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+  tag: 1.14.2
+`,
+			expectedKptfile: `apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: nginx
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/apply-setters:v0.1
+      configPath: setters.yaml
```
Comment (frankfarzan) [inline functions/go/put-setter-values/putsettervalues/putsettervalues_test.go:220]: have a test case for when setters.yaml is in a nested directory in the package.

**Review by frankfarzan — CHANGES_REQUESTED** [review]

**Comment by phanimarupaka** [comment]
@mengqiy verify-docs should not fail here. The function is supposed to be used by imperative invocation. PTAL.

**Comment by mengqiy** [comment]
We are going to have a meta function!

> @mengqiy verify-docs should not fail here. The function is supposed to be used by imperative invocation. PTAL.

I will fix it today.
